opampserver/pkg/server/handlers.go

```diff
@@ -146,13 +145,33 @@ func (c *ConnectionHandlers) PersistInstrumentationDeviceStatus(ctx context.Cont
 				Value: strValue,
 			})
 		}
+		dynamicOptions = append(dynamicOptions, instrumentation_instance.WithAttributes(identifyingAttributes, []odigosv1.Attribute{}))
+	}
+
+	isAgentDisconnect := message.AgentDisconnect != nil
```

Comment:
maybe better to check at the beginning of this func?
so it will return errors before creating the dynamicOptions

Reply:
moved to the top 👍